src/components/Chat/BottomSheetGif.js

```diff
@@ -48,15 +41,34 @@
     setIsLoading(false);
   };
 
+  const searchGIF = React.useCallback(
+    debounce((search) => {
+      onLoad(search);
+    }, 500),
+    []
+  );
+
+  const handleChangeText = (text) => {
+    setSearchText(text);
+    if (text?.trim()?.length > 2) {
+      searchGIF(text);
+    }
+  };
+
+  const handleOnClearText = () => {
+    setSearchText('');
+    searchGIF('');
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `searchGIF` function is debounced, which is good for performance. However, it's important to cancel any previous debounced calls when the component unmounts to avoid potential memory leaks or setting state on an unmounted component. You can do this by calling the `cancel` method provided by lodash's `debounce` function in a cleanup function within a `useEffect` hook.
```javascript
React.useEffect(() => {
  return () => {
    searchGIF.cancel();
  };
}, []);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->